.ci/integration_test

```diff
@@ -96,7 +99,14 @@ function setup_awscli() {
 
 function get_test_id() {
   git_commit=`git show -s --format="%H"`
-  export TEST_ID=etcdbr-test-${git_commit}
+  export TEST_ID=${TEST_ID_PREFIX}-${git_commit}
+  echo "Test id: ${TEST_ID}"
+}
+
+function get_latest_commit_test_id() {
+  git_commit_remote=`echo \`git ls-remote $URL HEAD\` | cut -d' ' -f 1`
```

Comment:
Won't this list all PR commits? 
What's the difference between `get_test_id` function and this one?

Reply:
It doesn't list PR commits. Just the latest commit on master branch on the remote repo. Reason for this is, we won't be able to run TM tests on every PR because it requires the image to be built from the current PR build. And for this to happen, the CI pipeline needs to execute the `publish` trait. After playing around with the CI definition, I realised there's no way to execute a step post `publish` trait (I may be wrong here, so would love your help on this). So I decided to populate test_id based on latest commit on remote repo's master branch. With the next PR for CI integration, I'll modify the test script to use the latest release version for the etcdbrctl image.

The other approach is to build a new docker image from the PR repo and push to registry, but that's quite tedious and unnecessary I felt. Because anyway we won't be running TM tests on every PR build, but rather on a head-update or even a cron basis. WDYS?

Reply:
> It doesn't list PR commits.
right. sorry my mistake. 

Ok got the intention behind running TM only on head update. That's totally fine.

but i still didn't understand, since the head will be checked out locally,  `get_test_id` function will give expected test_id and git_commit sha. isn't this function redundant then?

Reply:
Actually, I'm thinking we can't run it on head update as well, since in the pipeline we can't run a `step` *after* the `publish` trait. So I think it might be better to run the TM tests periodically, outside the scope of backup-restore CI pipeline, ie, the TM framework itself would trigger the etcd-backup-restore tests on a cron basis. In that case, since the test would have to pull the latest image from registry (which was built using the latest commit on the remote repo), I'm using a separate `get_latest_commit_test_id` for that. In fact, the `test_id` that we generate is only used to specify the bucket name. We could even generate a random ID for it, and since the test always cleans up the bucket, we wouldn't have to worry about `test_id` at all. WDYS?

Reply:
> Actually, I'm thinking we can't run it on head update as well, since in the pipeline we can't run a step after the publish trait.

We can run `step` after `publish`. You can make use of `depedes` condition mentioned at https://gardener.github.io/cc-utils/pipeline_step.html .

>  I'm using a separate get_latest_commit_test_id for that.

The test script is available locally means the repository is checked out locally. No, please remove   `get_latest_commit_test_id` and use `get_test_id`. Output of both is same in all scenario. But get_latest_commit_test_id unnecessary calls remote.

> the test_id that we generate is only used to specify the bucket name. We could even generate a random ID for it, and since the test always cleans up the bucket, we wouldn't have to worry about test_id at all.

Yes. But when the test scripts fails because of some bug or random event. Since test_id will map to commit, it will be easy to debug. That's why we didn't and should not go for random id in first place.

Reply:
> We can run step after publish. You can make use of depedes condition mentioned at https://gardener.github.io/cc-utils/pipeline_step.html .

`depends` only makes sure that `publish` trait runs after the dependency is satisfied, right?